spec/features/1_basic_spec.rb

```diff
@@ -0,0 +1,184 @@
+require "rails_helper"
+
+describe "User authentication" do
+  it "requires sign in before any action with the Devise `before_action :authenticate_user!` method", points: 2 do
+    visit "/movies/new"
+    current_url = page.current_path
+
+    expect(current_url).to eq(new_user_session_path),
```

Comment:
When Devise isn't installed this will fail since `new_user_session_path` isn't defined. That _won't_ make the custom error message display though, since the expectation never ran.
![image](https://user-images.githubusercontent.com/17581658/230464992-09201ddf-d77e-47a9-a3ba-925fc3a0b268.png)

If your intention was that the custom error message would be visible to students who _haven't_ created users,  then you should use the String path instead. Something like:

```rb
    expect(current_url).to eq("/users/sign_in"),
```

I also buy the idea that since the spec title already mentions Devise, student could infer that they need devise to make the test pass w/out the help of the error message.

Reply:
> I also buy the idea that since the spec title already mentions Devise, student could infer that they need devise to make the test pass w/out the help of the error message. It's up to you if you want to make this change.

Yeah I considered my use of the helper method there, but that's why I include Devise in the spec. The assumption is that they should have added devise and installed it, which will provide those helper methods.

Merging! Thanks for all your help!

Reply:
I remember now that students tend to work top down from the list of specs, so placing the Devise-required specs at the bottom will probably eliminate some friction. It's not a big deal though.

Reply:
Ah, sure. Just going to make that change directly on main.